acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/dsl/Transactions.java

```diff
@@ -41,15 +42,15 @@ public String submit(final Transaction transaction) throws IOException {
     return eth.sendTransaction(transaction);
   }
 
-  public JsonRpcErrorResponse submitExceptional(final Transaction transaction) throws IOException {
+  public SignerResponse<JsonRpcErrorResponse> submitExceptional(final Transaction transaction)
+      throws IOException {
     try {
       eth.sendTransaction(transaction);
       fail("Expecting exceptional response ");
       return null;
     } catch (final ClientConnectionException e) {
       LOG.info("ClientConnectionException with message: " + e.getMessage());
-      final String jsonBody = e.getMessage().substring(e.getMessage().indexOf("{"));
-      return Json.decodeValue(jsonBody, JsonRpcErrorResponse.class);
+      return Web3JErrorParser.parseConnectionException(e);
```

Comment:
Web3JErrorParser is really a property of Eth, as only Eth knows / controls how it calls the node i.e. if it changed from Web3j to say Vertx what would happen here?

Suggestion: create method on `Eth`:
`public JsonRpcErrorResponse parseException(final ClientConnectionException e)`

Reply:
If changed from web3j to vertx this would no longer work. Don't think it is however a property of Eth though, if anything it is probably a property of Signer.

Reply:
Had another look, agree Eth is probably the right place. So moved it there